doc/explanation/styling/design.md

```diff
@@ -0,0 +1,107 @@
+# Designs and Theming
+
+:::{versionadded} 1.0.0
+Designs and theming allow controlling the look and feel of individual components or your entire application.
+:::
+
+## `Design` class
+
+The `Design` class has the ability to apply styling and theming to components via three mechanisms:
+
+1. Loading external JS and CSS `resources`.
+2. Applying `modifiers` that directly override (or combine with) component parameter settings (including CSS stylesheets).
+3.
+
+Additionally a `Design` class always has an associated `Theme` which determines the color palette of the design. This also has the ability to:
+
+1. Apply a `base_css` (usually containing CSS variable definitions and inherited from the base class).
+2. Apply a `bokeh_theme` to override Bokeh component styling.
+3. Apply a `css` with specific CSS overrides for that theme.
+4. Apply its own set of `modifiers`.
+
+### Example: Bootstrap
+
+Let us work through the definition of the `Bootstrap` design step-by-step.
+
+#### Resources
+
+First of all, the `Bootstrap` design defines CSS and JS `resources` which load the bootstrap5 library:
+
+```python
+    _resources = {
+        'css': {
+            'bootstrap': CSS_URLS['bootstrap5']
+        },
+        'js': {
+            'bootstrap': JS_URLS['bootstrap5']
+        }
+    }
+```
+
+#### Modifiers
+
+Next we get to the `modifiers` which is where the meat of the styling definition lives:
+
+```python
+    modifiers = {
+        Accordion: {
+            'active_header_background': 'var(--bs-surface-bg)'
+        }
+		...
+        Tabulator: {
+            'theme': 'bootstrap4'
+        },
+        Viewable: {
+            'stylesheets': [Inherit, f'{CDN_DIST}bundled/theme/bootstrap.css']
```

Comment:
Would I put the Bokeh plots css fixes into fast.css or into some fast_holoviews.css file to use in the `stylesheets` for the `HoloViews` pane?

Reply:
>Would I put the Bokeh plots css fixes into fast.css or into some fast_holoviews.css file to use in the stylesheets for the HoloViews pane?

For now I put everything in one file but as this grows splitting them up makes sense.